### PR TITLE
initial react app prep

### PR DESCRIPTION
**Pull request by RobbMAlexander**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayCityData: false,
+      cityData: [],
+      userSearch: '',
+      errMsg: '',
+      errRender: false,
```
Comment (RivaD2) [inline src/App.js]: Good work on the lab so far! 

- Feedback here is to review state here as well as in `CityDisplay`, we have the same state in both components. When you can, grab a TA so they can help you decide where state is stored 👍

File: src/CityDisplay.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      errMsg: '',
+      errRender: false,
+
+    }
+  }
+
+  handleExplore = async () => {
+
+    try {
+      let cityResults = await axios.get('https://us1.locationiq.com/v1/search.php')
```
Comment (RivaD2) [inline src/CityDisplay.js]: - I see that we have this method which is grabbing city data as well as another method below that is grabbing city data
- Consider if the handleExplore should be responsible for handling some sort of event, in this case, when the user types in a city name, we will set the state using whatever the e.target.value is 👍 
- In this case, if we go this route, we can remove the try catch blocks and just have this method handle the event (then your `getCityData` will be responsible for grabbing data)

File: src/CityDisplay.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    let city = e.target.city.value
+    this.setState({
+      userSearch: e.target.city.value,
+    });
+    this.getCityData(city);
+  }
+
+  getCityData = async (city) => {
+    try {
+      let url = `https://us1.locationiq.com/v1/search.php?key=${process.env.local.REACT_APP_LOCATIONIQ_ACCESS_TOKEN}&q=${city}&format=json`;
```
Comment (RivaD2) [inline src/CityDisplay.js]: - Grab a TA if you can here too, where is `city` stored?

File: src/CityDisplay.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      let cityResults = await axios.get(url);
+
+      this.setState({
+        cityData: cityResults.data[0]
+      })
+    } catch (error) {
+      this.setState({
+        errRender: true,
+        errMsg: `An error has occurred: ${error.response.status}; ${error.response.data}`
```
Comment (RivaD2) [inline src/CityDisplay.js]: - line 56 is throwing an error, "cannot read properties of undefined reading 'status`"

File: src/CityDisplay.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        <header>
+          <h1>Learn About A City</h1>
+        </header>
+        <main>
+          <button onClick={this.handleExplore}>Display City Information</button>
+          {this.state.displayCityData ? cityRendered : ''}
+          {this.state.renderError && <p>{this.state.errMsg}</p>}
+          <form onSubmit={this.handleCitySubmit}>
+          <label>Input A City
+            <input name="city" type="text" />
```
Comment (RivaD2) [inline src/CityDisplay.js]: - This relates to my comments in your `handleExplore`. The input will need some sort of onChange prop that will make use of your handleExplore.

File: src/CityDisplay.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      })
+    } catch (error) {
+      this.setState({
+        errRender: true,
+        errMsg: `An error has occurred: ${error.response.status}; ${error.response.data}`
+      })
+    }
+  }
+
+  handleCitySubmit = e => {
```
Comment (KevLaMDev) [inline src/CityDisplay.js]: Your handleCitySubmit would be serve better as a callback func for an onInput attribute in your input tag. Doing so would update your state per user keystroke. Remember that using onInput/onChange means you access the user data as event.target.value not event.target.name.value

**Pull request merged**